build/source/engine/time_utils.f90

```diff
@@ -174,5 +175,92 @@ subroutine compjulday(iyyy,mm,id,ih,imin,dsec,&  ! input
 
  end subroutine compjulday
 
+ ! ***************************************************************************************
+ ! public subroutine compgregcal: convert julian day (units of days) to calendar date
+ ! source: https://en.wikipedia.org/wiki/Julian_day#Julian_or_Gregorian_calendar_from_Julian_day_number
+ ! ***************************************************************************************
+
+ subroutine compcalday(julday,                              & !input
+                       iyyy,mm,id,ih,imin,dsec,err,message)   !output
+ USE multiconst,only:secprmin  ! seconds in an (day, hour, minute)
+ implicit none
+
+ ! input variables	
+ real(dp), intent(in)          :: julday       ! julian day
+
+ ! output varibles
+ integer(i4b), intent(out)     :: iyyy         ! year
+ integer(i4b), intent(out)     :: mm           ! month
+ integer(i4b), intent(out)     :: id           ! day
+ integer(i4b), intent(out)     :: ih           ! hour
+ integer(i4b), intent(out)     :: imin         ! minute
+ real(dp),     intent(out)     :: dsec         ! seconds
+ integer(i4b), intent(out)     :: err          ! error code
+ character(*), intent(out)     :: message      ! error message
+
+ ! local parameters
+ integer(i4b),parameter       :: y = 4716
```

Comment:
This is a subroutine that Andy Neman extracted from wikipedia. The source is given in
`! source: https://en.wikipedia.org/wiki/Julian_day#Julian_or_Gregorian_calendar_from_Julian_day_number`
at the top of the subroutine.

To be honest, I don't think users need to understand why this works. However, we should clean up out time utilities -- and make things like this less cryptic.

I've added a note to issue #124 to improve time utilities.